modules/magniteAnalyticsAdapter.js

```diff
@@ -821,10 +830,16 @@ magniteAdapter.track = ({ eventType, args }) => {
         auctionEntry.floors.dealsEnforced = args.floorData.enforcements.floorDeals;
       }
 
-      // Log error if no matching bid!
+      // no-bid from server. Document it!
       if (!bid) {
-        logError(`${MODULE_NAME}: Could not find associated bid request for bid response with requestId: `, args.requestId);
-        break;
+        bid = adUnit.bids[generateUUID()] = {
+          bidder: 'mgnipbs',
+          source: 'server',
+          status: 'no-bid',
+          bidId: args.requestId,
+          clientLatencyMillis: args.timeToRespond || Date.now() - cache.auctions[args.auctionId].auction.auctionStart
+        };
+        return;
```

Comment:
For this scenario, I think we want something a bit different.

This is a valid bid response, so in this scenario we actually want to create a new bid in our adUnit

And then just let the rest of the bid_response processing logic do what it normally does:

So something like this:

```js
      // If we still do not have a matching bid object and this bidResponse has a seatBidId, lets create a default one
      if (!bid && args.seatBidId) {
        bid = adUnit.bids[args.seatBidId] = {
          bidder: args.bidderCode,
          bidId: args.seatBidId,
          status: 'no-bid',
          source: 'server',
          unknownBid: true // just so we can see how often this happens at log level
        }
      }

      // finally if still no bid just bail like before
      if (!bid) {
        logError(`${MODULE_NAME}: Could not find associated bid request for bid response with requestId: `, args.requestId);
        break;
      }
```

So we just have a new check before this `!bid` check to see if it falls into our unknown bid response from PBS scenario. And if it does, then we can just create a bid on our adUnit, set it to its defaults, and then let the regular bid_response processing go.

Reply:
Most of that sounds good to me, except letting regular bid response processing go. When that happens, we get this:
<img width="316" alt="Screenshot 2023-03-30 at 4 14 16 PM" src="https://user-images.githubusercontent.com/42588549/228976027-c33e919f-038f-45bb-909e-a432b04b7773.png">
Some of that could be weirdness with how we mock the getStatusCode function, but do we really want any of the rest besides latency? And if we want to avoid copy pasta with latency, I could just put that code into a function that gets called twice.